src/App.tsx

```diff
@@ -1,9 +1,223 @@
-import './App.scss';
+import { useMemo, useState, useContext } from 'react';
+import { useNavigate } from 'react-router';
+import {
+  Navigate,
+  Route,
+  Routes,
+  useLocation,
+  useSearchParams,
+} from 'react-router-dom';
+import { Context } from './Context';
+import { Header } from './Header';
+import { Navigation } from './Navigation';
+import { Home } from './Home';
+import { Products } from './Products';
+import { ProductDetails } from './ProductDetails';
+import { Favourites } from './Favourites';
+import { Cart } from './Cart';
+import { Footer } from './Footer';
+import { ProductTypes } from './types/productTypes';
 
-const App = () => (
-  <div className="App">
-    <h1>React Phone Catalog</h1>
-  </div>
-);
+const App = () => {
+  const { setQuery } = useContext(Context);
+  const { pathname } = useLocation();
+  const history = useNavigate();
+  const [searchParams, setSearchParams] = useSearchParams();
+  const [filterType, setFilterType] = useState('');
+  const page = searchParams.get('page' || '');
+  const perPage = searchParams.get('perPage' || '');
+  const sort = searchParams.get('sort' || '');
+  const filterQuery = searchParams.get('query' || '');
+  const activeProduct = JSON.parse(
+    localStorage.getItem('product') as string,
+  ) || null;
+  const updateSearch = (params: {
+    [key: string]: number[] | string[] | string | null
+  }) => {
+    Object.entries(params).forEach(([key, value]) => {
+      if (value === null) {
+        searchParams.delete(key);
+      } else if (Array.isArray(value)) {
+        searchParams.delete(key);
+
+        value.forEach(part => {
+          searchParams.append(key, String(part));
+        });
+      } else {
+        searchParams.set(key, value);
+      }
+
+      setSearchParams(searchParams);
+    });
+  };
+
+  useMemo(() => {
+    if (pathname === '/phones') {
+      setFilterType('phones');
+    } else if (pathname === '/tablets') {
+      setFilterType('tablets');
+    } else if (pathname === '/accessories') {
+      setFilterType('accessories');
+    } else if (pathname === '/favourites') {
+      setFilterType('favourites');
+    } else {
+      setFilterType('');
+    }
+
+    setQuery('');
+    updateSearch({ query: null });
+    history(-1);
+  }, [pathname]);
+
+  return (
+    <div className="app">
+      <Header
+        filterType={filterType}
+        filterQuery={filterQuery}
+        updateSearch={updateSearch}
+      />
+
+      <main style={{ flexGrow: 1 }}>
+        <Routes>
+          <Route
+            path="/"
+            element={(
+              <Home pathname={pathname} />
```

Comment:
why do we need to send pathname to each route, if we can easily get that pathname using some of react-router hook